### PR TITLE
Add side navigation.

### PR DESCRIPTION
**Pull request by nsthorat**
Fix some other issues along the way:
- Fix navigation between datasets/concepts directly by adding a {#key identifier} to each one, to destroy dom and recreate so the component is created again and reads the new context.
- Clean up settings a little bit for datasets, add delete dataset there.

https://huggingface.co/spaces/lilacai/nikhil_staging

https://github.com/lilacai/lilac/assets/1100749/9ebe0523-164e-47e8-ad08-b9e9ac3721d1

### CODE REVIEW

File: web/blueprint/src/lib/components/Navigation.svelte [hunk 1]
```diff
@@ -1,11 +1,109 @@
...
+  import NavigationGroup from './NavigationGroup.svelte';
+  import {Command, triggerCommand} from './commands/Commands.svelte';
+  import {hoverTooltip} from './common/HoverTooltip';
+
+  const authInfo = queryAuthInfo();
+  $: userId = $authInfo.data?.user?.id;
+  $: username = $authInfo.data?.user?.given_name;
+  $: canCreateDataset = $authInfo.data?.access.create_dataset;
+
+  $: urlHashContext = getUrlHashContext();
```
Comment (dsmilkov) [inline web/blueprint/src/lib/components/Navigation.svelte:22]: these 2 lines can be `const` since context is not reactive.

**Review by dsmilkov — APPROVED** [review]
amazing work

File: web/blueprint/src/lib/components/datasetView/Dataset.svelte [hunk 2]
```diff
@@ -96,123 +98,120 @@
...
-        class="!cursor-pointer"
-        on:click={() => {
-          const link = datasetLink(namespace, datasetName);
-          // Don't push a new state if the link matches.
-          if (link != location.pathname + location.hash) {
-            goto(link);
-          }
-        }}>{$datasetViewStore.datasetName}</Tag
-      >
+{#key identifier}
```
[the remark this answers is not in the document]
Comment (nsthorat) [reply web/blueprint/src/lib/components/datasetView/Dataset.svelte]: done

File: web/blueprint/src/routes/settings/+page.svelte [hunk 3]
```diff
@@ -13,24 +13,31 @@
...
+                <SelectItem value={emdField.name} />
+              {/each}
+            </Select>
+          {/if}
+        </div>
+      </section>
     </div>
-  </div>
-</Page>
+  </div></Page
```
[the remark this answers is not in the document]
Comment (nsthorat) [reply web/blueprint/src/routes/settings/+page.svelte]: done

**Pull request merged**